onnx/defs/__init__.py

```diff
@@ -120,3 +121,15 @@ def get_function_ops() -> List[OpSchema]:
 
 
 SchemaError = C.SchemaError
+
+def register_schema(op: OpSchema):
+    name: str = op.name
+    domain: str = op.domain
+    ver: int = op.since_version
+    assert ver > 0, f'OpSchema {domain}::{name} need positive version but got {ver}'
```

Comment:
I would test this behavior as well. Do we also need to check that the domain and name strings are valid?

Reply:
Looks like the version number can be anything: https://onnx.ai/onnx/repo-docs/IR.html#onnx-versioning

> Operator sets use a simple version number. Each operator set version represents a snapshot of the set of operators, and their semantics at a particular point in time.

> This specification does not provide guidance on what versioning scheme model producers should be using.

Reply:
Removed the assert.

If user passes a verison as `-1`, it will be [rewritten](https://github.com/onnx/onnx/blob/main/onnx/defs/schema.h#L1225) to '1' (`OpSchema::kUninitializedSinceVersion`) . So the negative version maybe an undefined behavior ?